Analyses/src/StepPointMCDumper_module.cc

```diff
@@ -189,12 +206,22 @@ namespace mu2e {
     const auto& ih = event.getValidHandle<StepPointMCCollection>(hitsInputTag_);
     for(const auto& i : *ih) {
 
-      hit_ = VDHit(toff_, i);
-
-      if(writeProperTime_) {
-        tau_ = SimParticleGetTau::calculate(i, spMCColls, decayOffCodes_);
+      if(writeVDHit_)hit_ = VDHit(i);
+      if(writeParticleState_) {
+	KinKal::VEC3 pos;
+	if(detectorSystem_)
+	  pos = KinKal::VEC3(det->toDetector(i.position()));
+	else
+	  pos = KinKal::VEC3(i.position());
```

Comment:
Suggest to initialize pos in declaration:

const KinKal::VEC3 pos = detectorSystem_ ? 
   KinKal::VEC3(det->toDetector(i.position())) :
   KinKal::VEC3(i.position());

Reply:
I have mixed feelings about inline conditionals, I use them, but I think younger coders find them hard to read.  I think there is no performance benefit with modern compilers.

Reply:
Hi Dave,
This is not so much about performance (even though it can only help), but about not having "dummy-initialized" or "wrong-initialized" variables.  WIthout inline conditionals one can not make pos here const, and having it const is a good safety and readability thing.  Once I see a variable declared const I don't have to worry about it having a different value couple code pages down; it helps.
About younger coders: the more they see that style the more used to it they will become.
Andrei